earthpy/spatial.py

```diff
@@ -74,7 +74,15 @@ def normalized_diff(b1, b2):
         raise ValueError("Both arrays should be of the same dimensions")
 
     n_diff = (b2 - b1) / (b2 + b1)
-    n_diff = np.ma.masked_invalid(n_diff)
+
+    # Mask invalid values
+    if np.isnan(n_diff).any():
+        n_diff = np.ma.masked_invalid(n_diff)
+
+        # Set inf values to nan
+        if np.isinf(n_diff).any():
```

Comment:
This is a nice adjustment to ensure those cells have a na placeholder rather than inf. 
I think we should update the docstring to acknowledge that this is what the function does in the case of a calculation that doesn't have a finite answer.

Reply:
The docstring has been updated. Also note that I changed the order and hierarchy of operations, so that inf values are set to nan first, and the mask is not contingent on existing nan values. This change ensures that the array is masked any time there are nan values (whether the nan values were in the input arrays or whether the nan values are the result of the function).